sdk/eventhub/azure-eventhub/azure/eventhub/_connection_string_parser.py

```diff
@@ -0,0 +1,109 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from ._common import DictMixin
+
+
+class EventHubConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+
+    def __init__(self, **kwargs):
+        self._fully_qualified_namespace = kwargs.pop("fully_qualified_namespace", None)
+        self._endpoint = kwargs.pop("endpoint", None)
+        self._event_hub_name = kwargs.pop("event_hub_name", None)
+        self._shared_access_signature = kwargs.pop("shared_access_signature", None)
+        self._shared_access_key_name = kwargs.pop("shared_access_key_name", None)
+        self._shared_access_key = kwargs.pop("shared_access_key", None)
+
+    @property
+    def fully_qualified_namespace(self):
+        """The fully qualified host name for the Event Hubs namespace.
+        The namespace format is: `<yournamespace>.servicebus.windows.net`.
+        """
+        return self._fully_qualified_namespace
+
+    @property
+    def endpoint(self):
+        """The endpoint for the Event Hubs resource. In the format sb://<FQDN>/"""
+        return self._endpoint
+
+    @property
+    def event_hub_name(self):
+        """Optional. The name of the Event Hub, represented by `EntityPath` in the connection string."""
+        return self._event_hub_name
+
+    @property
+    def shared_access_signature(self):
+        """
+        This can be provided instead of the shared_access_key_name and the shared_access_key.
+        """
+        return self._shared_access_signature
+
+    @property
+    def shared_access_key_name(self):
+        """
+        The name of the shared_access_key. This must be used along with the shared_access_key.
+        """
+        return self._shared_access_key_name
+
+    @property
+    def shared_access_key(self):
+        """
+        The shared_access_key can be used along with the shared_access_key_name as a credential.
+        """
+        return self._shared_access_key
+
+
+def parse_connection_string(conn_str):
+    # type(str) -> EventHubConnectionStringProperties
+    """Parse the connection string into a properties bag containing its component parts.
+
+    :param conn_str: The connection string that has to be parsed.
```

Comment:
make sure you double check that `conn_str` is used for conenction string everywhere in EH  - the naming should be consistent (if you already did it, ignore this)

Reply:
looks like `conn_str` is used in EH as well!